agent/doc/UML/hydra_lab_exception_monitor_design.puml

```diff
@@ -0,0 +1,56 @@
+@startuml exception_monitor
+interface ExceptionReporter {
+    + void reportException(Exception e)
+    + void reportException(String message, StackTraceElement[] stackTrace)
+}
+
+class ExceptionReporterManager implements ExceptionReporter {
+    List<ExceptionReporter> reporters
+
+    + void registerExceptionReporter(ExceptionReporter reporter)
+    + void reportException(Exception e)
+    + void reportException(String message, StackTraceElement[] stackTrace)
+}
+
+class AppCenterReporter extends ExceptionReporterManager {
+    AppCenterClient appCenterClient;
+    + void reportException(Exception e)
+    + void reportException(String message, StackTraceElement[] stackTrace)
+}
+
+
+class FileErrorLogger extends ExceptionReporterManager {
+    + void reportException(Exception e)
+    + void reportException(String message, StackTraceElement[] stackTrace)
+    + void writeToFile(String message, StackTraceElement[] stackTrace)
+}
+
+class AppCenterClient{
+    public HandledErrorLog createErrorLog(Thread thread, Exception testException, boolean fatal)
+}
+
+AppCenterClient <-- AppCenterReporter
+
+abstract class AppenderBase<E> {
```

Comment:
What's this for? Could you provide the full class name of AppenderBase?

Reply:
The fullname is ch.qos.logback.core.AppenderBase. It is used to transfer the error log.